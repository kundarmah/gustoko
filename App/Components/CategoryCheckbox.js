import PropTypes from 'prop-types';
import React from 'react';
import { StyleSheet } from 'react-native';
import { CheckBox } from 'react-native-elements'
import { Colors, Metrics } from '../Themes'

CategoryCheckbox.propTypes = {
    title: PropTypes.string.isRequired,
    checked: PropTypes.bool.isRequired,
    onPress: PropTypes.func.isRequired
};

export default function CategoryCheckbox(props) {
    return (
        <CheckBox
            center={true}
            title={props.title}
            iconLeft={true}
            textStyle={[styles.checkboxText, { color: props.checked ? '#fff' : Colors.primaryColor }]}
            containerStyle={[styles.checkboxContainer, { backgroundColor: props.checked ? Colors.primaryColor : 'transparent' }]}
            iconType='material'
            checkedIcon='done'
            uncheckedIcon='add'
            checkedColor='#fff'
            onPress={props.onPress}
            uncheckedColor={Colors.primaryColor}
            checked={props.checked}/>
    );
}

const styles = StyleSheet.create({
    checkboxText: {
        fontSize: Metrics.hp('2%'),
        fontWeight: 'normal'
    },
    checkboxContainer: {
        flex: 0,
        borderWidth: 1,
        borderColor: Colors.primaryColor,
        borderRadius: 20,
        marginLeft: 0,
        marginRight: 10,
        height: Metrics.hp('5%'),
        justifyContent: 'center'
    }
});
