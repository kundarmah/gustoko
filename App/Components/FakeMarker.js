import React, { Component } from 'react'
// import PropTypes from 'prop-types';
import { View, Text, Image } from 'react-native'
import styles from './Styles/FakeMarkerStyle'
import ImageResizeMode from 'react-native/Libraries/Image/ImageResizeMode'


export default class FakeMarker extends Component {
  // // Prop type warnings
  // static propTypes = {
  //   someProperty: PropTypes.object,
  //   someSetting: PropTypes.bool.isRequired,
  // }
  //
  // // Defaults for props
  // static defaultProps = {
  //   someSetting: false
  // }

  render () {
    const { profileImage } = this.props
    return (
      <View style={[styles.container]} pointerEvents="none">
        <View style={styles.pinTip}></View>
        <Image
          resizeMode={'center'}
          style={styles.marker}
          source={profileImage}
        />
      </View>
    )
  }
}
