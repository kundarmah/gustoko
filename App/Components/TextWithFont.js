import React, { PropTypes } from 'react'
import { TouchableOpacity, Text, Animated, View} from 'react-native'
import { Colors } from '../Themes'
import * as Animatable from 'react-native-animatable'

class RegularText extends React.Component {
  render () {
    return (
      <View {...this.props} style={[this.props.transparent && {backgroundColor: 'transparent'}, this.props.onTop && {zIndex: 9}]}>
        <Text style={[
          {fontFamily: this.props.fontFamily ? this.props.fontFamily : 'Muli', color: Colors.charcoal}, 
          this.props.error && {color: '#ff4444'}, 
          this.props.styles]}>
            {this.props.children}
        </Text>
      </View>
    )
  }
}

class AnimatedText extends React.Component {
  render () {
    return (
      <View style={[this.props.transparent && {backgroundColor: 'transparent'}, this.props.onTop && {zIndex: 9}]}>
        <Animatable.Text
          useNativeDriver={true}
          {...this.props}
          style={[
          {fontFamily: this.props.fontFamily ? this.props.fontFamily : 'Muli', color: Colors.charcoal}, 
          this.props.error && {color: '#ff4444'}, 
          this.props.styles]}>
            {this.props.children}
        </Animatable.Text>
      </View>
    )
  }
}

class BoldText extends React.Component {
  render () {
    return (
      <View>
        <Text style={[{fontFamily: 'Muli-Bold'}, this.props.styles]}>{this.props.children}</Text>
      </View>
    )
  }
}

export { RegularText, BoldText, AnimatedText }
