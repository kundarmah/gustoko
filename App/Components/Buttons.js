import PropTypes from 'prop-types'
import React from 'react'
import { TouchableOpacity } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import { Colors, Metrics } from '../Themes'
import Icon from 'react-native-vector-icons/FontAwesome'
import { RegularText } from '../Components/TextWithFont'

class ButtonGradient extends React.Component {
  static propTypes = {
    text: PropTypes.string,
    onPress: PropTypes.func,
    disabled: PropTypes.bool,
    gradientColors: PropTypes.array
  }

  render () {
    return (
      <LinearGradient colors={this.props.gradientColors || [Colors.esmyaGreen, Colors.esmyaBlue]} start={{x: 0, y: 0}} end={{x: 1.5, y: 0}} style={{borderRadius: this.props.radius}}>
      <TouchableOpacity 
        style={[{
              alignItems: 'center', 
              justifyContent: 'center',
              borderRadius: this.props.radius,
              width: '100%',
              height: '100%'},
              this.props.styles]} 
        onPress={this.props.onPress}
        disabled={this.props.disabled}>
        {this.props.icon}
        {this.props.children}
      </TouchableOpacity>
      </LinearGradient>
    )
  }
}

class ButtonInvert extends React.Component {
  render () {
    const {iconName, isSelected} = this.props
    return (
        <TouchableOpacity { ...this.props } style={{margin: 4, height: Metrics.hp('10%'), width: Metrics.wp('20%'), borderRadius: 4, justifyContent: 'center', alignItems: 'center', borderWidth: 0.5, borderColor: Colors.secondaryColor, backgroundColor: isSelected ? Colors.primaryColor : Colors.white}}>
          <Icon name={iconName ? iconName : "random"} size={Metrics.hp('3%')} color={isSelected ? Colors.white : Colors.primaryColor} />
          <RegularText styles={{color: isSelected ? Colors.white : Colors.secondaryColor, fontSize: Metrics.hp('1.64%')}}>{this.props.children}</RegularText>
        </TouchableOpacity>
    )
  }
}

export { ButtonGradient, ButtonInvert }