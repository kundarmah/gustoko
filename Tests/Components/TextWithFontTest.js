import 'react-native'
import React from 'react'
import { RegularText } from '../../App/Components/TextWithFont'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

test('RegularFont component renders correctly', () => {
  const tree = renderer.create(<RegularText />)
  expect(tree).toMatchSnapshot()
})

test('RegularFont component renders correctly with styles prop', () => {
  const tree = renderer.create(<RegularText style={{backgroundColor: '#000000'}}/>)
  expect(tree).toMatchSnapshot()
})
