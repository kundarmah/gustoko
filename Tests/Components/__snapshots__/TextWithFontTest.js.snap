// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegularFont component renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "Muli",
        },
        undefined,
        undefined,
      ]
    }
  />
</View>
`;

exports[`RegularFont component renders correctly 2`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "Muli",
        },
        undefined,
        undefined,
      ]
    }
  />
</View>
`;
